src/Auth.php

```diff
@@ -51,9 +51,19 @@ protected function getUser($uid)
             return false;
         }
 
-        if (!$user = User::getByUIDAndProvider($uid, $plugin->getProviderMachineName())) {
-            $info = self::getUserInfo($uid);
-
+        $user = User::getByUIDAndProvider($uid, $plugin->getProviderMachineName());
+        $info = self::getUserInfo($uid);
```

Comment:
You might want to consider caching this API response at least a few hours as the directory information likely won't change between logins on the same day.